core/src/repair/quic_endpoint.rs

```diff
@@ -698,6 +740,249 @@ impl<T> From<crossbeam_channel::SendError<T>> for Error {
     }
 }
 
+#[derive(Default)]
+struct RepairQuicStats {
+    connect_error_invalid_remote_address: AtomicU64,
+    connect_error_other: AtomicU64,
+    connect_error_too_many_connections: AtomicU64,
+    connection_error_application_closed: AtomicU64,
+    connection_error_connection_closed: AtomicU64,
+    connection_error_locally_closed: AtomicU64,
+    connection_error_reset: AtomicU64,
+    connection_error_timed_out: AtomicU64,
+    connection_error_transport_error: AtomicU64,
+    connection_error_version_mismatch: AtomicU64,
+    invalid_identity: AtomicU64,
+    no_response_received: AtomicU64,
+    read_to_end_error_connection_lost: AtomicU64,
+    read_to_end_error_illegal_ordered_read: AtomicU64,
+    read_to_end_error_reset: AtomicU64,
+    read_to_end_error_too_long: AtomicU64,
+    read_to_end_error_unknown_stream: AtomicU64,
+    read_to_end_error_zero_rtt_rejected: AtomicU64,
+    read_to_end_timeout: AtomicU64,
+    router_try_send_error_full: AtomicU64,
+    write_error_connection_lost: AtomicU64,
+    write_error_stopped: AtomicU64,
+    write_error_unknown_stream: AtomicU64,
+    write_error_zero_rtt_rejected: AtomicU64,
+}
+
+async fn report_metrics_task(name: &'static str, stats: Arc<RepairQuicStats>) {
+    loop {
+        tokio::time::sleep(Duration::from_secs(2)).await;
```

Comment:
Can we define the "2" as a constant on the top?

Reply:
done.